### PR TITLE
Feat/mumbai deployment

### PR DESCRIPTION
**Pull request by MrDeadCe11**
adding deployments to main branch.

### CODE REVIEW

File: packages/contracts/tasks/deploy.ts [hunk 1]
```diff
@@ -1,60 +1,92 @@
...
     };
 
     //TODO Correct arguments for Mumbai/Polygon
     const constructorArguments: Record<Contracts, string[]> = {
-      QuadraticFundingCurator: ["0x45cf9Ba12b43F6c8B7148E06A6f84c5B9ad3Dd44", admin.address],
+      QuadraticFundingCurator: ['0x45cf9Ba12b43F6c8B7148E06A6f84c5B9ad3Dd44', admin.address],
       QuadraticVoteCollectModule: [
-        "0x60Ae865ee4C725cd04353b5AAb364553f56ceF82",
-        "0x1353aAdfE5FeD85382826757A95DE908bd21C4f9",
+        lensPolygonAddresses.lensHubImplementation,
```
Comment (bitbeckers) [inline packages/contracts/tasks/deploy.ts:29]: This blocks the use of optional parameters. Preferably, check at the beginning whether they've been provided and take action accordingly

File: packages/contracts/tasks/deploy.ts [hunk 2]
```diff
@@ -1,60 +1,92 @@
...
+    }
+
     for (const [name, contract] of Object.entries(contracts)) {
       console.log(`Starting deployment of ${name}`);
       const factory = contract;
-      const constructorArgs = Object.entries(constructorArguments).find((entry) => entry[0] === name)?.[1];
-      console.log(`Constructor arguments: ${constructorArgs}`);
+      let constructorArgs;
+
+      if (hre.network.name == 'polygon-mainnet') {
```
Comment (bitbeckers) [inline packages/contracts/tasks/deploy.ts:51]: This is kinda iffy. You're repeating the same logic, so it seems like the constructorArguments should be able to handle all networks insteads of creating separate steps

File: packages/contracts/tasks/deploy.ts [hunk 3]
```diff
@@ -1,60 +1,92 @@
...
+          if (code === '0x') {
             console.log(`${instance.name} contract deployment has not completed. waiting to verify...`);
             await instance.instance?.deployed();
           }
-          await hre.run("verify:verify", {
+
+          await hre.run('verify:verify', {
             address: instance.address,
-            constructorArgs,
+            contract: `contracts/${name}.sol:${name}`,
```
Comment (bitbeckers) [inline packages/contracts/tasks/deploy.ts:84]: This contract should already be available in this script. Why are you loading from a file?

Comment (MrDeadCe11) [reply packages/contracts/tasks/deploy.ts:84]: this is where it broke on the cache issue.  loading the specific contract fixed the caching error.

File: packages/contracts/tasks/deploy.ts (hunk 2 above)
Comment (MrDeadCe11) [reply packages/contracts/tasks/deploy.ts:51]: Deleted!

File: packages/contracts/tasks/deploy.ts (hunk 1 above)
Comment (MrDeadCe11) [reply packages/contracts/tasks/deploy.ts:29]: ok, here's my 1st attempt.  not the most clever but it works.  what do you think?

File: packages/contracts/tasks/deploy.ts (hunk 3 above)
Comment (bitbeckers) [reply packages/contracts/tasks/deploy.ts:77]: Removed it to find out what happens. This is not needed in any other project I set up with these scripts.

**Review by bitbeckers — APPROVED** [review]

**Pull request merged**